src/plugin/plugin.rs

```diff
@@ -195,51 +178,47 @@ where
         // the `RapierConfiguration` if it already exists.
         app.init_resource::<RapierConfiguration>();
 
-        app.insert_resource(SimulationToRenderTime::default())
-            .insert_resource(RapierContext {
-                physics_scale: self.physics_scale,
-                ..Default::default()
-            })
-            .insert_resource(Events::<CollisionEvent>::default())
-            .insert_resource(Events::<ContactForceEvent>::default())
-            .insert_resource(Events::<MassModifiedEvent>::default());
+        app.insert_resource(RapierContext {
+            physics_scale: self.physics_scale,
+            ..Default::default()
+        })
+        .insert_resource(Events::<CollisionEvent>::default())
+        .insert_resource(Events::<ContactForceEvent>::default())
+        .insert_resource(Events::<MassModifiedEvent>::default());
 
         // Add each set as necessary
         if self.default_system_setup {
+            // NOTE: SyncBackend must be run before any StepSimulation.
             app.configure_sets(
-                self.schedule,
-                (
-                    PhysicsSet::SyncBackend,
-                    PhysicsSet::StepSimulation,
-                    PhysicsSet::Writeback,
-                )
-                    .chain()
+                FixedUpdate,
+                PhysicsSet::SyncBackend.before(PhysicsSet::StepSimulation),
+            );
+            app.configure_sets(
+                FixedUpdate,
+                PhysicsSet::StepSimulation
+                    .after(PhysicsSet::SyncBackend)
                     .before(TransformSystem::TransformPropagate),
             );
+            app.configure_sets(
+                Update,
```

Comment:
This runs normally in Update, but why you want run this as late as possible? In my opinion running this as early as possible after FixedUpdate loop will be the best idea, because this do not create additional time for applying a data.